### PR TITLE
add bi-directional friendship

### PR DESCRIPTION
**Pull request by sz16900**
- Implement friendships with 2 rows for mutual friendship
- Add bi-directional friendship (i.e: User A is a friend with User B / User B is a friend with User A)
- Add Unit tests for friendship

### CODE REVIEW

File: app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@ def update
     friendship = Friendship.find_by_id(fs.ids)
     friendship.confirmed = true
     friendship.save
+    reverse_fs = Friendship.new(user_id: current_user.id, friend_id: params[:id], confirmed: true)
+    reverse_fs.save
```
Comment (euqueme) [inline app/controllers/friendships_controller.rb]: - [x] This save could be moved to the Friendship model using an after_create [callback ](https://guides.rubyonrails.org/active_record_callbacks.html#multiple-conditions-for-callbacks) so you don't do it within the controller

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @sz16900 and @nriqu322 :lollipop: :balloon: 

### I think you've achieved what was requested for this milestone, :medal_sports: :medal_sports:  let's just add the test cases that properly validate the reverse friendships :construction: :construction_worker: :construction_worker: 

* [x] **Unit and Integration test** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)) please cover the following test cases so it's not possible to have duplicate or self friendships from the model level:

- Unit
  - validates against duplicate friendship
  - validates against duplicate reverse-friendship
  - validates against self-friendship
- Integration
  - See posts from the user and his or her friends in the Timeline (in the previous milestone it was optional but in this milestone is mandatory so let's add an integration test case to validate it)

### That's it, team! :raised_hands: 

### Please :pray: do the changes :technologist: :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team ,  You have done a great job with this project. :clap:
#### You can go ahead and merge !!!
